src/dotty/tools/dotc/core/pickling/TreePickler.scala

```diff
@@ -531,8 +535,16 @@ class TreePickler(pickler: TastyPickler) {
       withLength { pickleType(ann.symbol.typeRef); pickleTree(ann.tree) }
     }
 
+    def updateMapWithDeltas[T](mp: collection.mutable.Map[T, Addr]) = {
+      mp.map{
+        case (key, addr) => (key, adjusted(addr))
```

Comment:
Updating a mutable data structure during iteration doesn't work in general. I guess this is safe because you don't change the key set, but I would be cautious about using the same approach in general.

If you were to use an `mutable.AnyRefMap` rather than a `mutable.Map` (which is a good idea for performance reasons anyway), you can use `mp.transformValues(adjusted)`.